Xamarin.Forms.Core/Behaviors/BehaviorBase.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using Xamarin.Forms;
+
+namespace Xamarin.Forms.Behaviors
+{
+    public class BehaviorBase<T> : Behavior<T> where T : BindableObject
+	{
+		public T AssociatedObject { get; private set; }
```

Comment:
this is all wrong. Behaviors are meant to be declared in Styles, and have shared instances. So your behavior should keep track of an AssociatedObject, but an array of AssociatedObjects.

and then you need to keep a kind of proxy for the bindable context thing, so the bindings are applied.

I'm guilty of this, I'm the one who made Behavior inherit from BindableObject

Reply:
@StephaneDelcroix In my opinion, that's not an issue.
Isn't the same problem with any other object type which is not meant to work in Style? Developers should be aware the Style is shared and doesn't work as a template. For example, do triggers work when put in a Style? 

In WPF there's a [x:Shared](https://docs.microsoft.com/en-us/dotnet/framework/xaml-services/x-shared-attribute) attribute , see https://stackoverflow.com/questions/22321966/interaction-triggers-in-style-in-resourcedictionary-wpf/22329342
Also See this interesting thread for using Behavior in Style in WPF: https://stackoverflow.com/questions/1647815/how-to-add-a-blend-behavior-in-a-style-setter

So I think having Behavior<T> like how David implemented it is OK.

Reply:
Triggers and Behaviors work when set in a Style, yes.

We could support `x:Shared`, please open a enhancement request for it.